sources/osgShader/Compiler.js

```diff
@@ -1158,9 +1160,9 @@ define( [
 
             var inputWeights = this.getOrCreateAttribute( 'vec4', 'Weights' );
             var inputBones = this.getOrCreateAttribute( 'vec4', 'Bones' );
-            var matrixPalette = this.getOrCreateUniform( 'vec4', 'uBones', this._animation.getBoneUniformSize() );
+            var matrixPalette = this.getOrCreateUniform( 'vec4', 'uBones', this._skinAttribute.getBoneUniformSize() );
```

Comment:
that would be more coherent to call the variable _skinningAttribute , if there are more modif that would great to fix it else we can maybe skip it